### PR TITLE
React refresh tweaks

### PR DESCRIPTION
**Pull request by DeMoorJasper**
## What kind of change does this PR introduce?

Minor improvements to the React Refresh logic and memory usage in React apps.

## What is the current behavior?

Wrap code is pretty large

## What is the new behavior?

Wrap code is largely encapsulated in helper :D

## Checklist

- [x] Documentation
- [x] Testing 
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/5669/commits/0de372816e0914c65c18df7ab185b481c7d5a0b3 is at https://pr5669.build.csb.dev/s/new.

File: packages/app/src/sandbox/eval/transpilers/react/refresh-transpiler.ts [hunk 1]
```diff
@@ -121,67 +112,76 @@ var registerExportsForReactRefresh = (moduleExports, moduleID) => {
...
  */
 const getWrapperCode = (sourceCode: string) =>
-  `var prevRefreshReg=window.$RefreshReg$,prevRefreshSig=window.$RefreshSig$,RefreshRuntime=require("react-refresh/runtime");window.$RefreshReg$=(e,r)=>{const s=module.id+" "+r;RefreshRuntime.register(e,s)},window.$RefreshSig$=RefreshRuntime.createSignatureFunctionForTransform;try{${sourceCode}
-}finally{window.$RefreshReg$=prevRefreshReg,window.$RefreshSig$=prevRefreshSig}const _csbRefreshUtils=require("${HELPER_PATH}"),isHotUpdate=!!module.hot.data,prevExports=isHotUpdate?module.hot.data.prevExports:null;if(_csbRefreshUtils.isReactRefreshBoundary)if(_csbRefreshUtils.isReactRefreshBoundary(module.exports)){_csbRefreshUtils.registerExportsForReactRefresh(module.exports,module.id);const e={...module.exports};module.hot.dispose((function(r){r.prevExports=e})),isHotUpdate&&_csbRefreshUtils.shouldInvalidateReactRefreshBoundary(prevExports,e)?module.hot.invalidate():module.hot.accept(),_csbRefreshUtils.enqueueUpdate()}else isHotUpdate&&_csbRefreshUtils.isReactRefreshBoundary(prevExports)&&module.hot.invalidate();`;
+  `var _csbRefreshUtils = require("${HELPER_PATH}");
```
Comment (DeMoorJasper) [inline packages/app/src/sandbox/eval/transpilers/react/refresh-transpiler.ts:174]: This kinda messes with sourcemaps, so don't merge this yet

**Review by CompuIves — APPROVED** [review]
Nice!!

File: packages/app/src/sandbox/eval/transpilers/react/refresh-transpiler.ts [hunk 2]
```diff
@@ -121,67 +121,69 @@ var registerExportsForReactRefresh = (moduleExports, moduleID) => {
...
   isReactRefreshBoundary,
   registerExportsForReactRefresh,
-  shouldInvalidateReactRefreshBoundary
+  shouldInvalidateReactRefreshBoundary,
+  prelude,
+  postlude,
 };
 `.trim();
 
-// const getWrapperCode = (sourceCode: string) =>
```
Comment (CompuIves) [inline packages/app/src/sandbox/eval/transpilers/react/refresh-transpiler.ts:132]: Ah, so you moved the wrapper code to the prelude and the postlude?

**Review by CompuIves — COMMENTED** [review]
I was a bit too early with approving, I read this code wrong first. I have one question first!

File: packages/app/src/sandbox/eval/transpilers/react/refresh-transpiler.ts (hunk 2 above)
Comment (DeMoorJasper) [reply packages/app/src/sandbox/eval/transpilers/react/refresh-transpiler.ts:132]: Yes, that's pretty much it. There's some other small tweaks to readability but no logic changes besides that

**Comment by CompuIves** [comment]
Great!! This looks very good!

**Pull request merged**